internal/api/oidc/sign/signer.go

```diff
@@ -0,0 +1,56 @@
+package sign
+
+import (
+	"context"
+	"sync"
+
+	"github.com/go-jose/go-jose/v4"
+
+	"github.com/zitadel/zitadel/internal/telemetry/tracing"
+	"github.com/zitadel/zitadel/internal/zerrors"
+)
+
+// SignerFunc is a getter function that allows add-hoc retrieval of the instance's signer.
+type SignerFunc func(ctx context.Context) (jose.Signer, jose.SignatureAlgorithm, error)
+
+// GetSignerOnce returns a function which retrieves the instance's signer from the database once.
+// Repeated calls of the returned function return the same results.
+func GetSignerOnce(
+	getActiveSigningWebKey func(ctx context.Context) (*jose.JSONWebKey, error),
+) SignerFunc {
+	var (
+		once    sync.Once
+		signer  jose.Signer
+		signAlg jose.SignatureAlgorithm
+		err     error
+	)
+	return func(ctx context.Context) (jose.Signer, jose.SignatureAlgorithm, error) {
+		once.Do(func() {
```

Comment:
If this is run once and `getActiveSigningWebKey` is returning an object subject to eventual consistency, don't you risk of returning the wrong object?